packages/odf/components/mcg/CreateObjectBucketClaim.tsx

```diff
@@ -51,6 +78,124 @@ export const CreateOBCForm: React.FC<CreateOBCFormProps> = (props) => {
     dispatch({ type: 'setProvisioner', name: sc?.provisioner });
   };
 
+  const generateOBCNameByNamespaceStore = (name: string): string => {
+    const base64Encoded = btoa(name).substring(0, 10);
+    return `destination-bucket-${name}-${base64Encoded}`.toLowerCase();
+  };
+
+  const generateBucketClassNameByNamespaceStore = (name: string): string => {
+    const base64Encoded = btoa(name).substring(0, 10);
+    return `generic-bucketclass-${name}-${base64Encoded}`.toLowerCase();
+  };
+
+  const createReplicationRulesAndStringify = (
+    data: ReplicationRuleFormData[],
+    namespace: string
+  ): string => {
+    const rules: Array<OBCReplicationRules> = data.map((rule) => {
+      const repRule: OBCReplicationRules = {
+        ruleId: `rule-${rule.ruleNumber}`,
+        obName: `obc-${namespace}-${generateOBCNameByNamespaceStore(
+          rule.namespaceStore
+        )}`,
+        filter: {
+          prefix: rule.prefix,
+        },
+      };
+      return repRule;
+    });
```

Comment:
nits: u can skip this comment, if u want to... not really needed...
```suggestion
    const rules: Array<OBCReplicationRules> = data.map((rule) => ({
        ruleId: `rule-${rule.ruleNumber}`,
        obName: `obc-${namespace}-${generateOBCNameByNamespaceStore(
          rule.namespaceStore
        )}`,
        filter: {
          prefix: rule.prefix,
        },
      }));
```